### PR TITLE
HTCONDOR-2112 add QUERY_MULTIPLE_ADS and QUERY_MULTIPLE_PVT_ADS to co…

### PR DESCRIPTION
**Pull request by johnkn**
…llector and query object

<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Review by JaimeFrey — COMMENTED** [review]
In the negotiator, the new code doesn't set ATTR_SEND_PRIVATE_ATTRIBUTES for the privateQuery. That means a current collector won't send the ClaimId attributes.

**Comment by JaimeFrey** [comment]
Ignore my comment about ATTR_SEND_PRIVATE_ATTRIBUTES in the privateQuery. I see now the code in the collector where it always sends the private attributes of the startd private ads.

File: src/condor_collector.V6/collector.cpp [hunk 1]
```diff
@@ -874,104 +1068,162 @@ int CollectorDaemon::receive_query_cedar_worker_thread(void *in_query_entry, Str
...
+			sock->encode();
+			sending = true;
+		}
+		if (whichAds == STARTD_PVT_AD) { filter_private_attrs = false; }
+
+		bool evaluate_projection = proj_is_expr;
+		classad::References * whitelist = proj.empty() ? nullptr : &proj;
+		if (query_entry->is_multi) {
+			std::string tagattr(query_entry->adt[ix].tag); tagattr += ATTR_PROJECTION;
+			if (query->LookupString(tagattr, projection)) {
```
Comment (JaimeFrey) [inline src/condor_collector.V6/collector.cpp:1146]: If a type-specific projection attribute needs collector-side evaluation, it's silently ignored.

Comment (johnkn) [reply src/condor_collector.V6/collector.cpp:1146]: yes, type specific projections can only be strings in the current code.

**Review by JaimeFrey — APPROVED** [review]
LGTM

**Pull request merged**